opt_einsum/paths.py

```diff
@@ -84,6 +81,56 @@ def optimal(input_sets, output_set, idx_dict, memory_limit):
     return path
 
 
+def _parse_possible_contraction(positions, input_sets, output_set, idx_dict,
+                                memory_limit, path_cost, naive_cost):
+    # Find the contraction
+    contract = helpers.find_contraction(positions, input_sets, output_set)
+    idx_result, new_input_sets, idx_removed, idx_contract = contract
+
+    # Sieve the results based on memory_limit
+    if helpers.compute_size_by_dict(idx_result, idx_dict) > memory_limit:
+        return None
+
+    # Build sort tuple
+    removed_size = helpers.compute_size_by_dict(idx_removed, idx_dict)
+    cost = helpers.flop_count(idx_contract, idx_removed,
+                              len(positions), idx_dict)
+    sort = (-removed_size, cost)
+
+    # Sieve based on total cost as well
+    if (path_cost + cost) > naive_cost:
+        return None
+
+    # Add contraction to possible choices
+    return [sort, positions, new_input_sets]
+
+
+def _update_other_results(results, best, input_sets):
+    """Update the positions and provisional input_sets of ``results`` based on
+    performing the contraction result ``best``.
+    """
```

Comment:
Do we have any checks to make sure the number of indices passed to einsum does not exceed what einsum can handle?

Reply:
So ``_np_einsum`` ensures that only good characters are used for each contraction call to einsum. If there were more than 52 supplied to an individual einsum call then that would be a problem but this would I guess require a 52+ dimensional tensor if all contractions are pairwise!